kerykeion/charts/chart_drawer.py

```
@@ -4710,14 +4800,29 @@ def _write_svg_to_disk(
+        chartname = output_directory / f"{self._sanitize_output_basename(str(base_name))}.svg"
+
+        # Defense in depth: ensure the resolved target stays inside the
+        # resolved output directory (covers symlinks and exotic inputs).
+        resolved_directory = output_directory.resolve()
+        resolved_chartname = chartname.resolve()
+        try:
+            resolved_chartname.relative_to(resolved_directory)
+        except ValueError:
+            raise KerykeionException(
+                f"Refusing to write SVG outside the output directory: {resolved_chartname} is not inside {resolved_directory}."
+            )
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)